crates/parser/src/format/sanitize/datetime.rs

```diff
@@ -0,0 +1,99 @@
+use crate::{ParseConfig, Output, format::ParseResult, ParseError};
+use chrono::{DateTime, FixedOffset, SecondsFormat};
+use chrono_tz::Tz;
+use serde_json::Value;
+
+struct DatetimeSanitizer {
+    from: Output,
+    default_timezone: Tz,
+}
+
+const NAIVE_FORMATS: [&'static str; 4] = [
+    "%Y-%m-%dT%H:%M:%S",
+    "%Y-%m-%dT%H:%M:%S%.3f",
+    "%Y-%m-%d %H:%M:%S%.3f",
+    "%Y-%m-%d %H:%M:%S",
+];
+
+const FORMATS: [&'static str; 2] = [
+    "%Y-%m-%d %H:%M:%S%.3f%:z",
```

Comment:
> Another thing is that I don't see the T here. I see that some test cases have it, so I'm wondering if those cases are passing because we're just passing them through as is. Is that intentional and important? If so, it's deserving of a comment.

Yes we just pass through RFC3339 as is, it doesn;t need us to parse it and format it again

Reply:
> I also wonder if chrono's built-in rfc3339 format would handle all of those cases without needing to try both of these formats. Is there a reason not to use that?

RFC3339 requires the `T`, here we are trying to handle the case where the `T` is missing, which is not valid RFC3339